src/FsCheck/FSharp.ArbMap.fs

```diff
@@ -28,7 +28,23 @@ module ArbMap =
     /// parameters of type Arbitrary<'T> to obtain them. Arbitrary instance of the correct type are automatically
     /// injected on construction of a particular type.
     let mergeWith<'TArb> (existingMap: IArbMap) = mergeWithType typeof<'TArb> existingMap
+    
+    /// Return a new Type to Arbitrary map that merges the existing map with the Arbitrary<'T> returned by the given function.
+    let mergeFactory (factory: unit -> Arbitrary<'b>) (existingMap: IArbMap) = 
+        (existingMap :?> ArbMap).MergeFactory(Func<unit,Arbitrary<'b>>(factory)) :> IArbMap
```

Comment:
I think `MergeFactory` should go on `IArbMap` instead of the casts.

Reply:
Makes sense to me

Reply:
Well, something to consider real quick. I implemented it this way to mimic mergeWithType. I thought, perhaps, we didn't want to expose any modification on IArbMap. IArbMap is currently a read-only abstraction, and MergeFactory would change that. I'm not sure how much it matters, since all the modification is exposed by proxy...